felix/collector/collector.go

```diff
@@ -306,12 +306,18 @@ func (c *collector) getDataAndUpdateEndpoints(t tuple.Tuple, expired bool, packe
 	dstEp := c.lookupEndpoint(t.Src, t.Dst)
 	dstEpIsNotLocal := dstEp == nil || !dstEp.IsLocal
 
-	if !okData {
+	if !exists {
 		// For new entries, check that at least one of the endpoints is local.
 		if srcEpIsNotLocal && dstEpIsNotLocal {
 			return nil
 		}
 
+		// Ignore HEP reporters.
+		if (srcEp != nil && srcEp.IsLocal && srcEp.IsHostEndpoint()) ||
```

Comment:
I think this change is right.  I'm wondering if it should also be reflected in line 338 though?  I'm not totally sure what the possible circumstances are for an endpoint to change - maybe it's a timing thing on deletion?

Thinking through it, the only problematic case is if somehow a pod could turn into a HEP.  I don't think that's actually possible though.

Reply:
For an execution to reach line 338, first the endpoint should exist in the ep map, by reaching line 323, to execute `updateEpStatsCache()`, and then in the next iteration it might reach 338. 

So even, if endpoint change is possible, collector will first need to add it to the ep map first.